rust/abacus-base/src/contract_sync/outbox.rs

```diff
@@ -79,43 +84,25 @@ where
         //    Note this means we only handle this case upon observing messages in some range [C,D]
         //    that indicate a previously indexed range may have missed some messages.
         tokio::spawn(async move {
-            let mut from = db
-                .retrieve_latest_valid_message_range_start_block()
-                .unwrap_or(config_from);
-
-            let mut last_valid_range_start_block = from;
+            let mut cursor = cursor.await?;
 
-            info!(from = from, "[Messages]: resuming indexer from latest valid message range start block");
+            let start_block = cursor.current_position();
+            let mut last_valid_range_start_block = start_block;
+            info!(from = start_block, "[Messages]: resuming indexer from latest valid message range start block");
+            indexed_height.set(start_block as i64);
 
-            indexed_height.set(from as i64);
             loop {
-                sleep(Duration::from_secs(5)).await;
-
-                // Only index blocks considered final.
-                // If there's an error getting the block number, just start the loop over
-                let Ok(tip) = indexer.get_finalized_block_number().await else {
-                    continue;
-                };
-                if tip <= from {
-                    // Sleep if caught up to tip
-                    sleep(Duration::from_secs(10)).await;
-                    continue;
-                }
-
-                // Index the chunk_size, capping at the tip.
-                let to = min(tip, from + chunk_size);
-
-                // Still search the full-size chunk size to possibly catch events that nodes have dropped "close to the tip"
-                let full_chunk_from = to.checked_sub(chunk_size).unwrap_or_default();
+                let start_block = cursor.current_position();
+                let Ok((from, to)) = cursor.next_range().await else { continue };
```

Comment:
should we log if there's an error? I think we'll get some logs if we're using QuorumProvider but because that probably shouldn't be assumed, I think logging here & in the other places we get errors from a `next_range` call would be helpful for potential debugging